src/main/kotlin/org/jetbrains/research/testspark/tools/kex/generation/KexProcessManager.kt

```diff
@@ -0,0 +1,350 @@
+                    }
+                }
+
+                override fun processTerminated(event: ProcessEvent) {
+                    log.info("Process terminated with exit code: ${event.exitCode}")
+                    log.info("Output from Kex stdout:\n ${output.stdout}")
+                    log.info("Output from Kex stderr:\n ${output.stderr}")
+                }
+            })
+
+            handler.startNotify()
+
+            // Wait for the process to complete with a timeout
+            if (!handler.waitFor(kexProcessTimeout)) {
+                handler.destroyProcess()
+                throw IOException("Process timed out and was terminated")
+            }
+        } catch (e: IOException) {
+            e.printStackTrace()
+            log.error("Error running KEX process", e)
+        }
+    }
+
+    private fun getHelperClassBody(testCode: String) =
+        (
+            StaticJavaParser.parse(testCode).findFirst(ClassOrInterfaceDeclaration::class.java)
+                .get()
+            ).toString().substringAfter('{').substringBeforeLast('}')
+
+    private fun extractTestMethod(file: File?, id: Int): MethodDeclaration {
+        val compilationUnit = StaticJavaParser.parse(file)
+        // Kex generates a Class for each test case. Each class has exactly two methods:
+        // at index 1 an @Before annotated method
+        // at index 2 an @Test annotated method
+        // merge the two into a single method
+        val methods = compilationUnit.findAll(MethodDeclaration::class.java)
+        val methodStmts = methods.map { it.body.map { it.statements }.orElse(NodeList()) }
+
+        val fields = NodeList(
+            compilationUnit.findAll(FieldDeclaration::class.java)
+                .filterNot { it.isPublic } // drops the timeout
+                .map { it.toString() }
+                .map { StaticJavaParser.parseStatement(it) },
+        )
+        fields.addAll(methodStmts[1])
+        fields.addAll(methodStmts[2])
+        methods[2].setBody(BlockStmt(fields))
+        methods[2].setName("test$id")
+        return methods[2]
+    }
+
+    /**
+     * Ensures that the necessary files for Kex execution exist. If the files already exist, this method will return true.
+     * If any required file is missing, this method will attempt to download the necessary files.
+     *
+     * @return False if kex doesn't exist and files couldn't be downloaded
+     */
+    private fun ensureKexExists(): Boolean {
+        val kexDir = File(kexHome)
+        if (!kexDir.exists()) {
+            kexDir.mkdirs()
+        }
+
+        // All jars in runtime deps are also required
+        // There are more files on top of this which I'm not aware of so the kex zip on github just has the whole project
+        // Importantly this list has kaxExecPath which contains a version number. So if the number changes the project will be updated with a download
+        val requiredFiles = listOf(
```

Comment:
File names/paths inside kex structure that we depend on also should me extracted as static constants when possible.

Reply:
The problem here is I don't know what the required files are. The comment explains the reason for the check, is simply for updating based on the version number of the kex jar. So i can check for just that here? In which case `kexExecPath` is already a constant field.
Otherwise could you let me know what other files kex needs to run?

Reply:
Well, we need to make the purpose of this check explicit. If the only thing that matters here is the version check, we can discard everything else and just compare the versions of the jar files with the expected version.